services/wallet/reader.go

```diff
@@ -210,13 +197,17 @@ func (r *Reader) GetWalletToken(ctx context.Context, addresses []common.Address)
 		return nil
 	})
 
+	clients, err := r.rpcClient.EthClients(chainIDs)
+	if err != nil {
+		return nil, err
+	}
 	group.Add(func(parent context.Context) error {
-		clients, err := r.rpcClient.EthClients(chainIDs)
-		if err != nil {
-			return err
-		}
 		balances, err = r.tokenManager.GetBalancesByChain(ctx, clients, addresses, tokenAddresses)
 		if err != nil {
+			for _, client := range clients {
+				client.SetIsConnected(false)
+			}
+			log.Info("tokenManager.GetBalancesByChain err", err)
```

Comment:
Do we want to keep this?

Reply:
(And have it with log level `Info`)

Reply:
yeah, similar to other places, we can tweak later